src/DoctrineObject.php

```diff
@@ -314,6 +318,10 @@ public function hydrateValue(string $name, $value, ?array $data = null)
             return null;
         }
 
+        if (PHP_VERSION_ID >= 80100 && interface_exists(BackedEnum::class) && $value instanceof BackedEnum) {
```

Comment:
IMO just `$value instanceof BackedEnum` - the other checks are redundant.

Reply:
I agree they are regarding runtime, but Psalm complains then:

```
ERROR: UndefinedClass - src/DoctrineObject.php:319:30 - Class, interface or enum named Doctrine\Laminas\Hydrator\BackedEnum does not exist (see https://psalm
.dev/019)
        if (interface_exists(BackedEnum::class) && $value instanceof BackedEnum) {
```

Any suggestions? Otherwise, I'd stick with the current check.

Reply:
Nevermind, got it working (caching issue) 🙈